pyproximal/proximal/ETP.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+from scipy.special import lambertw
+
+from pyproximal.ProxOperator import _check_tau
+from pyproximal import ProxOperator
+
+
+class ETP(ProxOperator):
+    r"""Exponential-type penalty (ETP).
+
+    Parameters
+    ----------
+    sigma : :obj:`float`
+        Regularization parameter.
+    gamma : :obj:`float`, optional
+        Regularization parameter. Default is 1.0.
+
+    Notes
+    -----
+    The exponential-type penalty is defined as
```

Comment:
Also following the standard of pyproximal do string, I suggest to put the Mathematical expression in the main description whilst have an explanation of its proximal in notes

Reply:
OK

Reply:
I find it good to have some math of the proximal even though it's concise as when you write it you know it is obvious but some time later may not be and also for users may not be :)

Reply:
OK, I wrote something out, but since it doesn't have a closed-form expression I didn't know how to handle it. See if you like it.

Reply:
I like it, this looks very good :)

Reply:
OK, I think it is ready to merge then